src/main/resources/XWiki/Macros/Team.xml

```diff
@@ -0,0 +1,1407 @@
+        <number>3</number>
+        <prettyName>Macro description</prettyName>
+        <rows>5</rows>
+        <size>40</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </description>
+      <id>
+        <disabled>0</disabled>
+        <name>id</name>
+        <number>1</number>
+        <prettyName>Macro id</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </id>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Macro name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <priority>
+        <disabled>0</disabled>
+        <name>priority</name>
+        <number>11</number>
+        <numberType>integer</numberType>
+        <prettyName>Priority</prettyName>
+        <size>10</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.NumberClass</classType>
+      </priority>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#set($discard = $xwiki.ssx.use('xwiki:XWiki.Macros.Team'))
+#set($discard = $xwiki.jsx.use('xwiki:XWiki.Macros.Team'))
+#set($pictureList = {})
+#macro(avatarInitials $name $size $letterAvatarBgColor $letterAvatarFontColor)
+  &lt;span class="xwikiteam-avatar xwikiteam-avatar-initials" style="height:${size}px; width:${size}px; border-radius: ${size}px; background-color: $letterAvatarBgColor"&gt;
```

Comment:
I don't understand the question. The values need to be escaped depending on how they are used. The escaping needs to be done inside the macro code, because we can't trust the values passed by the user as macro parameters.

Reply:
avatarInitials is an internal macro that is called with value that have already been escaped. I guess eventually having avatarInitials as an autonomous macro would be cool so escaping inside the macro would be worth it, however I need to avoid double escaping.

I guess I can do this.

Reply:
I did and there was no issues with double escaping in the end. Values passed to this macro that risk double escaping have no legal values where double escaping would pose a problem.